### ISSUE TITLE
Multibyte chars indicies

### ISSUE THREAD

**Issue by l4l** [Blakeinstein/fuse-rust#8]
It seems that search indices don't work with multibyte chars (or at least not clear how to use these):

```rust
let needle = "f";
let s = "®f∮";

let fuse = Fuse::default();
let pat = fuse.create_pattern(needle);
let x = fuse.search(pat.as_ref(), s).unwrap();

let r = &x.ranges[0];

// # The following assertion fails with:
//
// panicked at 'byte index 1 is not a char boundary; it is inside '®' (bytes 0..2) of `®f∮`'
assert_eq!(s.split_at(r.start).1.split_at(r.end).0, needle);

// # Using char indices also fails:
//
// panicked at 'assertion failed: `(left == right)`
//  left: `"f∮"`,
// right: `"f"`'
assert_eq!(
    s.chars().skip(r.start).take(r.len()).collect::<String>(),
    needle
);
```

**Comment by Blakeinstein** [comment]
I have a general idea as to how I can go about fixing this. I will look into it soon.

**Comment by Blakeinstein** [comment]
I probably should just use graphemes at this point... anyways, I am partially done with a fix, atleast the ranges returned should be byte indices.. I will fix the search part in a while. Sorry for the delay!

**Comment by l4l** [comment]
Yeah, right, using single codepoints is not enough, since a glyph might consists of multiple characters:
`assert_eq!(vec!['o', '\u{308}'], "ö".chars().collect::<Vec<char>>())`

**Comment by Blakeinstein** [comment]
I thought of using graphemes instead, but then that would be too much work for someone trying to use the library.
I wanted the ranges to be simply useable by `string[r.start..r.end]`.

I hadnt thought of multi-byte chars while writing this, as while learning rust, I assumed that string chars would have consistent sizes. Anyways, I am working on a rewrite with multibyte chars and working with `char_indices()`.

With rewrites, searching for grapheme chars that are in two parts like è would match with e, assuming I understand how its encoded.

**Comment by Blakeinstein** [comment]
Should be resolved with v0.2. I just ended up using byte indexing to match bytes instead of actual characters. 
I know there is a multibyte character that can be indexed separately, but I am an unsure which one. Shouldn't cause any unusual behavior but I don't have enough test cases to check.

I apologize for the delay, and hope that you raise issues regarding any further complications.

**Comment by l4l** [comment]
Sure, yeah. Nevertheless I'm not sure it still fits for my case but anyway the fix addresses the issue in general, so I'm closing the issue. Thanks for your effort!
